config/routes.rb

```diff
@@ -3,4 +3,9 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  get "/ingredients", to: "ingredients#index"
+
+  get "/recipes/:id", to: "recipes#show"
+  patch "/recipes/:id", to: "recipes#update"
```

Comment:
Overall good restful routes! I don't agree with line 9 for the 2nd extension, but I think you were halfway through this one.